src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -586,11 +590,6 @@ impl Step for Rustdoc {
             features.as_slice(),
         );
 
-        // If the rustdoc output is piped to e.g. `head -n1` we want the process
-        // to be killed, rather than having an error bubble up and cause a
-        // panic.
-        cargo.rustflag("-Zon-broken-pipe=kill");
```

Comment:
Looks like this was added in https://github.com/rust-lang/rust/commit/cde0cde151f30e07170d0f3caa721c92bebd7afe, regressing the fix I made in https://github.com/rust-lang/rust/pull/123192. Seems likely this will regress again.

Can you add a comment here saying to *not* add any rustflags, as that will cause undesired rebuilds?

Reply:
> Looks like this was added in https://github.com/rust-lang/rust/commit/cde0cde151f30e07170d0f3caa721c92bebd7afe, regressing the fix I made in https://github.com/rust-lang/rust/pull/123192. Seems likely this will regress again.

It shouldn't regress. We still set `-Zon-broken-pipe=kill` for rustdoc.

> Can you add a comment here saying to not add any rustflags, as that will cause undesired rebuilds?

We need to duplicate that comment for every `impl Step for $tool` to make it clear, which wouldn't be not good enough either (because people can miss comments sometimes). As I mentioned in https://github.com/rust-lang/rust/issues/123177#issuecomment-2188335301, I will create a follow-up PR later this week to have this automatically checked. So the test can ensure tools never break the build cache again.

Reply:
:shrug: IMO one can't have too many comments (or at least, that is very hard). Tests are nice but they only catch things after someone already made their PR, when CI runs. And tests can't cover all combinations of tools either.

Reply:
We definitely need a change that ensures we never end up breaking the cache again (for any tool not just rustdoc).

Added a comment for rustdoc as it's harmless to have.